pandas/parser.pyx

```diff
@@ -1208,7 +1208,16 @@ cdef class TextReader:
             else:
                 return None
 
-class CParserError(Exception):
+
```

Comment:
There are two options.  As I wrote above, the "new" code I added is just a mirror of the C engine code, which throws a `CParserError` when the header indices don't match where the reader stops.  So either one: you change `CParserError` to a more generic sounding `ParserError` or you add `PythonParserError`.